Generator/Dependency.ts

```diff
@@ -0,0 +1,49 @@
+import { Cell } from "./Cell";
+import { SimpleSolver } from "./SimpleSolver";
+import { SingleStrategies } from "./SingleStrategies";
+import { copy2dCellArray } from "./Sudoku";
+
+/**
+ * Contains function to return dependency score of a Sudoku board to be used to calculate difficulty.
+ * Inspired by https://www.fi.muni.cz/~xpelanek/publications/flairs-sudoku.pdf
+ * Calculates difficulty based on the dependencies of the steps used to solve the board.
+ * Calculates dependency score by counting how many applications of simple strategies are available at each step.
+ * So if a puzzle has an average of 6 naked/hidden singles are each step it is easier than one where you have to search for only a couple available at each step.
```

Comment:
Maybe we can reword? 
```
So if a puzzle has an average of 6 naked/hidden singles are each step it is easier than one where you have to search for only a couple available at each step.
``` 
So if a puzzle has an average of 6 naked/hidden singles for each step then it is easier than a puzzle where there is only an average of two naked/hidden singles each step.